drivers/modem/gsm_ppp.c

```diff
@@ -571,10 +583,14 @@ static void set_ppp_carrier_on(struct gsm_modem *gsm)
 	}
 }
 
-static void query_rssi(struct gsm_modem *gsm)
+static void query_rssi(struct gsm_modem *gsm, bool lock)
```

Comment:
Instead of a parameter to the function, imho it is more readable to have a locking version of the function.
So something like
```
static void query_rssi_lock(struct gsm_modem *gsm)
{
     gsm_ppp_lock(gsm);
     query_rssi(gsm);
     gsm_ppp_unlock(gsm);
}
```
This way user does not really need to remember what the extra bool parameter is for.